clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -215,6 +215,12 @@ protected void onJoinComplete(int generation,
 
         // update partition assignment
         subscriptions.assignFromSubscribed(assignment.partitions());
+        this.joinedSubscription = subscriptions.subscription();
```

Comment:
@hachikuji Here is my understanding, and correct me if I'm wrong:

1. In `assignFromSubscribed`, we will only change the `subscription` if it is regex pattern and the collected topics as in `topicsToSubscribe` is a superset of `this.subscription`.

2. Besides this added line, `joinedSubscription` is only set to `this.subscription` before sending the join group request, and the `this.subscription` itself is only modified either inside `subscribe()` calls on inside the previously line if the assigned topics is a super set of the subscription itself. 

So given these two, I think this line will only update `joinedSubscription` with those topics that were added to the assignment by the leader If that does happen, right?